webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/services/reports/ExecutionBatchTest.scala

```diff
@@ -1026,6 +1033,90 @@ class ExecutionBatchTest extends Specification {
     }
   }
 
+  /*
+   * A rule with:
+   * - one historical technique/directive without reportId: d1
+   * - one ncf technique with reportId: gm1 and gm2 are in block (block1), gm3 is under root
+   * Data from a real cases https://issues.rudder.io/issues/20804
+   *
+   *  r = 32377fd7-02fd-43d0-aab7-28460a91347b (RuleId(RuleUid(uuid),Revision(default)))
+   *  d1 = DirectiveId(DirectiveUid(50a48619-5f04-489b-85f4-80f46968ac40),Revision(default))
+   *  d2 = DirectiveId(DirectiveUid(24ea2fd9-93e0-4de0-8ceb-773e86554e76),Revision(default))
+   *  nodeId = NodeId(root)
+   *  message0 = No post-modification script was defined
+   *  reportid1 = 4c4858a1-ed05-4a74-9362-624d5475d451
+   *  value1 = /bin/true
+   *  message1 = Execute the command /bin/true was repaired
+   *  reportid2 = 10957b22-20e5-43c3-8ef2-aa5b666b3bd9
+   *  value2 = /tmp/rudder-file-gm2
+   *  message2 = Presence of file /tmp/rudder-file-gm2 was correct
+   *  reportid3 = e0a496c2-14a5-4cd2-81bf-8d4630549b41
+   *  value3 = /tmp/rudder-file-gm3
+   *  message3 = Presence of file /tmp/rudder-file-gm3 was correct
+   *
+   *  t1 = 2022-02-23T11:40:30.000+01:00
+   *  t2 = 2022-02-23T11:40:23.000+01:00
+   *
```

Comment:
these logs are a bit verbose, but we can alsmost copy/past then in the test to check what is going wrong